symplyphysics/laws/optics/radius_newton_ring_from_wavelength_order_radius_curvature_refractive_index.py

```diff
@@ -0,0 +1,57 @@
+from sympy import (Eq, solve, sqrt)
+from symplyphysics import (
+    units,
+    Quantity,
+    Symbol,
+    print_expression,
+    validate_input,
+    validate_output,
+    dimensionless,
+)
+
+# Description
+## Newton's rings are annular interference maxima and minima that appear around the point of contact between
+## a convex lens and a plane—parallel plate when light passes through the lens and plate.
+## The order of interference indicates the number of whole wavelengths. When the difference in the path of the
+## beam reflected from the convex surface of the lens at the glass-air boundary and the beam reflected from the
+## plate at the air-glass boundary is equal to an entire wave "n", then these two waves hit the observation
+## point with the same faces and demonstrate interference.
+
+## Law is: r = sqrt(k * R * L / n), where
+## r - radius of Newton's ring,
+## k - order of interference,
+## R - radius of curvature of lens
+## L - wavelength in a vacuum,
+## n - refractive index of medium between lens and plate.
+
+radius = Symbol("radius", units.length)
+
+wavelength = Symbol("wavelength", units.length)
+double_order_interference = Symbol("double_order_interference", dimensionless)
```

Comment:
Why `double`? I do not see anything about double order in the description.

Reply:
> Why `double`? I do not see anything about double order in the description.

This was discussed in another task. Here, as in the other task, it is assumed to use the formula for both dark and light rings. The formula itself is for dark rings, but if you substitute fractional values instead of integers instead of k, then you can get the radii of light rings. Therefore, as I understand it, you suggested such a name.

https://github.com/blackyblack/symplyphysics/pull/309#discussion_r1469985707
https://github.com/blackyblack/symplyphysics/pull/311#discussion_r1468933432
https://github.com/blackyblack/symplyphysics/pull/311#discussion_r1470143962

Reply:
No, I suggested this name because it indicates double amount of half waves. This way we could avoid fractional order of interference.